gallery/visualtests/nxMultiFileUpload.js

```diff
@@ -0,0 +1,488 @@
+      expect(textContent2).toMatch('2.0 kB');
+
+      await dismissResultingDialog(async () => {
+        await dismissBtn1.click();
+      });
+
+      [selectedFile1] = await waitAndGetElements(`${statefulExampleSelector} .nx-selected-file:nth-child(1)`);
+      textContent1 = await selectedFile1.evaluate(e => e.textContent);
+
+      // confirm the correct file was removed
+      expect(await isInDocument(selectedFile2)).toBe(false);
+      expect(await isInDocument(selectedFile1)).toBe(true);
+      expect(textContent1).toMatch('2.0 kB');
+    });
+
+    it('sets the no File Selected message when the last selected file is removed', async function() {
+      await openFileChooser([files.bytes]);
+
+      const [selectedFile, dismissBtn] = await waitAndGetElements(
+          `${statefulExampleSelector} .nx-selected-file`,
+          `${statefulExampleSelector} .nx-selected-file .nx-selected-file__dismiss-btn`
+      );
+
+      await dismissResultingDialog(async () => {
+        await dismissBtn.click();
+      });
+
+      const [noFileSelectedMessage] = await waitAndGetElements('.nx-file-upload__no-file-message'),
+          textContent = await noFileSelectedMessage.evaluate(e => e.textContent);
+
+      expect(await isInDocument(selectedFile)).toBe(false);
+      expect(await isInDocument(noFileSelectedMessage)).toBe(true);
+      expect(textContent).toBe('No file selected');
+    });
+
+    describe('focus behavior', function() {
+      it('sets focus on the appropriate dismiss button once a file is removed', async function() {
+        await openFileChooser([files.bytes, files.kilobytes, files.megabytes, files.gigabytes]);
+
+        let [selectedFile1, dismissBtn1, selectedFile2, dismissBtn2, selectedFile3, dismissBtn3, selectedFile4] =
+          await waitAndGetElements(
+              `${statefulExampleSelector} .nx-selected-file:nth-child(1)`,
+              `${statefulExampleSelector} .nx-selected-file:nth-child(1) .nx-selected-file__dismiss-btn`,
+              `${statefulExampleSelector} .nx-selected-file:nth-child(2)`,
+              `${statefulExampleSelector} .nx-selected-file:nth-child(2) .nx-selected-file__dismiss-btn`,
+              `${statefulExampleSelector} .nx-selected-file:nth-child(3)`,
+              `${statefulExampleSelector} .nx-selected-file:nth-child(3) .nx-selected-file__dismiss-btn`,
+              `${statefulExampleSelector} .nx-selected-file:nth-child(4)`
+          );
+
+        // certify all 4 files are indeed uploaded
+        expect(await isInDocument(selectedFile1)).toBe(true);
+        expect(await isInDocument(selectedFile2)).toBe(true);
+        expect(await isInDocument(selectedFile3)).toBe(true);
+        expect(await isInDocument(selectedFile4)).toBe(true);
+
+        // remove the first file
+        await dismissResultingDialog(async () => {
+          await dismissBtn1.click();
+        });
+
+        [selectedFile1, dismissBtn1, selectedFile2, dismissBtn2, selectedFile3, dismissBtn3] =
```

Comment:
Rather than re-querying all of these, you should be able to reuse the values they all got set to in the previous `waitAndGetElements`, and assert that the first one is no longer in the doc

Reply:
> Rather than re-querying all of these, you should be able to reuse the values they all got set to in the previous `waitAndGetElements`, and assert that the first one is no longer in the doc

Thanks for the suggestion, updated here: [57108b0](https://github.com/sonatype/sonatype-react-shared-components/pull/850/commits/57108b09e47b964d69d11f67d2982d3408d63672)